libretroshare/src/retroshare/rsjsonapi.h

```diff
@@ -0,0 +1,118 @@
+
+	virtual void setBindingAddress(const std::string& address) =0;
+	virtual void setListeningPort(uint16_t port) =0;
+	virtual uint16_t listeningPort() const =0;
+
+    virtual void connectToConfigManager(p3ConfigMgr *cfgmgr)=0;
+
+    virtual void registerResourceProvider(const JsonApiResourceProvider *)=0;
+    virtual void unregisterResourceProvider(const JsonApiResourceProvider *)=0;
+    virtual bool hasResourceProvider(const JsonApiResourceProvider *)=0;
+
+	/**
+	 * @brief Get status of the json api server
+	 * @jsonapi{development}
+	 * @return the status picked in the enum JSONAPI_STATUS_UNKNOWN/RUNNING/NOT_RUNNING
+	 */
+	virtual int status() =0;
+
+    //=============================================================================================//
+    //                        API methods that are also accessible through http                    //
+    //=============================================================================================//
+	/**
+	 * @brief This function should be used by JSON API clients that aren't
+	 * authenticated yet, to ask their token to be authorized, the success or
+	 * failure will depend on mNewAccessRequestCallback return value, and it
+	 * will likely need human user interaction in the process.
+	 * @jsonapi{development,unauthenticated}
+	 * @param[in] token token to autorize
+	 * @return true if authorization succeded, false otherwise.
+	 */
+	virtual bool requestNewTokenAutorization(const std::string& token)=0;
+
+    //=============================================================================================//
+    //                                    Utility methods                                          //
+    //=============================================================================================//
+
+    static bool parseToken(const std::string& clear_token,std::string& user,std::string& passwd);
+
+    //=============================================================================================//
+    //                      API methods that SHOULD NOT be accessible through http                 //
+    //=============================================================================================//
+
+    //////////////// @Gio: The methods below should not be accessible from the API server !
+    ///
+	/**
+	 * @brief Add new auth (user,passwd) token to the authorized set, creating the token user:passwd internally.
+	 * @param[in] alphanumeric_user username to autorize decoded
+	 * @param[in] alphanumeric_passwd passwd to autorize decoded
+	 * @return true if the token has been added to authorized, false if error occurred
+	 */
+	virtual bool authorizeUser(const std::string& alphanumeric_user,const std::string& alphanumeric_passwd)=0;
+
+	/**
+	 * @brief Revoke given auth token
+	 * @param[in] user par of the decoded token
+	 * @return true if the token has been revoked, false otherwise
+	 */
+	virtual bool revokeAuthToken(const std::string& user)=0;
```

Comment:
this should be accessible from JSON API, and they where authenticated too, think of Android the only way you have to interact with the library is through JSON API